provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -538,6 +575,54 @@ func megabytesToKiB(diskMiB uint64) uint64 {
 	return diskMiB * 1024
 }
 
+func (c *Client) buildDiskLocator(
+	ctx context.Context,
+	args CreateVirtualMachineParams,
+	srcVM *object.VirtualMachine,
+	datastore types.ManagedObjectReference,
+) ([]types.VirtualMachineRelocateSpecDiskLocator, error) {
+
+	templateDisks, err := c.getDisks(ctx, srcVM)
+	if err != nil {
+		return nil, errors.Annotatef(err, "source VM disks")
+	}
+
+	var scrub bool
+	var thinProvision bool
+
+	switch args.DiskProvisioningType {
+	case DiskTypeThick:
+		scrub = false
+		thinProvision = false
+	case DiskTypeThickEagerZero:
+		scrub = true
+		thinProvision = false
+	case DiskTypeThin:
```

Comment:
```
default:
    // If no disk provisioning type is specified, fall back to
    // thin disk provisioning.
    fallthrough
case DiskTypeThin:
```

Reply:
Indeed! Thanks for this.

One small note for posterity, the default will be thick with eager zeros due to security concerns. See comments above for each disk provisioning type for details.